JADE.io (Australia).js

```diff
@@ -0,0 +1,819 @@
+ * Handles common formats like "13 August 2025" and falls back to Date parsing
+ *
+ * @param {string} d - Date string to normalise
+ * @returns {string} - ISO date format or original string if parsing fails
+ */
+function normaliseDate(d) {
+	if (!d) return "";
+	// Handle "13 August 2025" and similar AU formats
+	const m = String(d).match(/(\d{1,2})\s+([A-Za-z]+)\s+([12]\d{3})/);
+	if (m) {
+		const day = m[1].padStart(2, "0");
+		const months = {
+			january: "01", february: "02", march: "03", april: "04", may: "05", june: "06",
+			july: "07", august: "08", september: "09", october: "10", november: "11", december: "12"
+		};
+		const mon = months[m[2].toLowerCase()];
+		if (mon) return `${m[3]}-${mon}-${day}`;
+	}
+	// Fallback to Date parsing (may be locale dependent)
+	try {
+		const dt = new Date(d);
+		if (!isNaN(dt.getTime())) return dt.toISOString().slice(0, 10);
+	}
+	catch (e) {
+		// Date parsing failed - return original
+	}
+	return d;
+}
+
+/**
+ * Locates PDF download links on the page for attachment
+ * Searches for links with .pdf extension or download attributes
+ *
+ * @param {Document} doc - HTML document object
+ * @returns {string|null} - PDF URL if found, null otherwise
+ */
+function findPDFLink(doc) {
+	const link = doc.querySelector('a[href$=".pdf"], a[href*="download"][href*="pdf"], a[download$=".pdf"]');
+	return link ? link.href : null;
+}
+
+/**
+ * Preferred law report series ordering for parallel citations
+ * Based on Australian Guide to Legal Citation (AGLC) hierarchy
+ * Used to determine which parallel citation to use in main reporter fields
+ *
+ * Priority: Commonwealth Law Reports (CLR) > Australian Law Journal Reports (ALJR) > etc.
+ */
+const LAW_REPORT_PRIORITY = [
+	'CLR',
+	'ALJR',
+	'ALR',
+	'FCR',
+	'NSWLR',
+	'VR',
+	'Qd R',
+	'SASR',
+	'WAR',
+	'NTLR',
+	'ACTR',
+	'FamLR',
+	'FLR',
+	'IR',
+	'A Crim R',
+	'ACSR',
+	'ATPR',
+	'ATC'
+];
+
+/**
+ * Finds all parallel law report citations in the document
+ * Searches for patterns like "274 CLR 450" throughout the page text
+ *
+ * @param {Document} doc - HTML document object
+ * @returns {Array} - Array of {volume, reporter, page} citation objects, de-duplicated
+ */
+function findLawReportCitations(doc) {
+	const repPattern = LAW_REPORT_PRIORITY
+		.map(r => r.replace(/\s+/g, "\\s+"))
+		.join("|");
+	const re = new RegExp(`\\b(\\d{1,4})\\s+(${repPattern})\\s+(\\d{1,5})\\b`, 'g');
+
+	function scanAll(s) {
+		if (!s) return [];
+		const out = [];
+		let m;
+		while ((m = re.exec(s))) {
+			out.push({ volume: m[1], reporter: m[2].replace(/\s+/g, ' '), page: m[3] });
+		}
+		return out;
+	}
+
+	// Aggregate from focused sources only to avoid cited-case noise
+	let results = [];
+	results = results.concat(scanAll(getMeta(doc, "og:title", "property")));
+	results = results.concat(scanAll(getMeta(doc, "og:description", "property")));
+	const els = doc.querySelectorAll(
+		".ribbon-citation, li.ribbon-citation, .citation, .citations, .case-citation"
```

Comment:
Redundant selector:

```suggestion
		".ribbon-citation, .citation, .citations, .case-citation"
```

And do all of these actually occur?